compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -14267,11 +14344,21 @@ private STNode parseAnonFuncExprOrTypedBPWithFuncType(List<STNode> qualifiers) {
      */
     private STNode parseAnonFuncExprOrFuncTypeDesc(List<STNode> qualifiers) {
         startContext(ParserRuleContext.FUNC_TYPE_DESC_OR_ANON_FUNC);
-        STNode qualifierList = createFuncTypeQualNodeList(qualifiers);
+        STNode qualifierList;
         STNode functionKeyword = parseFunctionKeyword();
-        STNode funcSignature = parseFuncSignature(true);
+        STNode funcSignature;
+        if (peek().kind == SyntaxKind.OPEN_PAREN_TOKEN) {
+            funcSignature = parseFuncSignature(true);
+            qualifierList = createFuncTypeQualNodeList(qualifiers, true);
+        } else {
+            funcSignature = STNodeFactory.createEmptyNode();
+            qualifierList = createFuncTypeQualNodeList(qualifiers, false);
+        }
```

Comment:
else means it should be type desc right? If so we can directly parse rest for type desc and return.